### ISSUE TITLE
Clean up CMakeLists.txt

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#188]
The file starts to get messy. I think we should try to use separate files for the submodules.

Tutorial: http://johnlamp.net/cmake-tutorial-4-libraries-and-subdirectories.html
Files: https://github.com/acgetchell/cmake-tutorial

ToDoCore could easily be replaced by CoolPropCore :smile:

**Comment by ibell** [comment]
Bump?

**Comment by ibell** [comment]
Shall we delay this one?  I have some ideas to clean it up a bit anyway, like making all the SWIG-based wrappers use a common core.

**Comment by henningjp** [comment]
Mathcad Prime is the "forward progress" version of Mathcad, but Mathcad 15 will be around for more than a couple of years yet, so I can't pull that out until Prime catches up with 15 and surpasses it enough that it becomes the obvious choice for users.  So unfortunately both of those wrappers are needed.  I'm happy to test, or if you've got a template I can work from, I can work on those wrappers myself.

**Comment by thorade** [comment]
Not sure how much work it would save, but for the Python wrappers one could consider dropping support for versions that have reached end of life (i.e. 3.3):
https://docs.python.org/devguide/#status-of-python-branches
https://pypi.python.org/pypi/CoolProp
Also, one could consider dropping 32bit support, at least for the Py3 versions!?

And it is important to add a source release to PyPI in addition to the wheels.

**Comment by jowr** [comment]
Thanks for the inputs. I find myself using 32bit Python every now and then, especially when interacting with old proprietary DLLs that only exist in 32bit and 16bit. I guess we should keep 32bit and 64bit for Python 2.7, 3.4, 3.5 and 3.6 as well as the source release.

**Comment by ibell** [comment]
Agreed.  I need it for refprop for instance and testing the 32 bit dll

On Nov 30, 2017 8:54 AM, "Jorrit Wronski" <notifications@github.com> wrote:

Thanks for the inputs. I find myself using 32bit Python every now and then,
especially when interacting with old proprietary DLLs that only exist in
32bit and 16bit. I guess we should keep 32bit and 64bit for Python 2.7,
3.4, 3.5 and 3.6 as well as the source release.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/CoolProp/CoolProp/issues/188#issuecomment-348194099>,
or mute the thread

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
